kong/init.lua

```diff
@@ -766,8 +725,120 @@ function Kong.access()
   -- we intent to proxy, though balancer may fail on that
   ctx.KONG_PROXIED = true
 
-  if kong.ctx.core.buffered_proxying then
-    return buffered_proxy(ctx)
+  return Kong.response()
+end
+
+do
+  local HTTP_METHODS = {
+    GET       = ngx.HTTP_GET,
+    HEAD      = ngx.HTTP_HEAD,
+    PUT       = ngx.HTTP_PUT,
+    POST      = ngx.HTTP_POST,
+    DELETE    = ngx.HTTP_DELETE,
+    OPTIONS   = ngx.HTTP_OPTIONS,
+    MKCOL     = ngx.HTTP_MKCOL,
+    COPY      = ngx.HTTP_COPY,
+    MOVE      = ngx.HTTP_MOVE,
+    PROPFIND  = ngx.HTTP_PROPFIND,
+    PROPPATCH = ngx.HTTP_PROPPATCH,
+    LOCK      = ngx.HTTP_LOCK,
+    UNLOCK    = ngx.HTTP_UNLOCK,
+    PATCH     = ngx.HTTP_PATCH,
+    TRACE     = ngx.HTTP_TRACE,
+  }
+
+  function Kong.response()
+    if not kong.ctx.core.buffered_proxying then
+      return
+    end
+
+
+    -- from buffered_proxy
+
+    local ctx = ngx.ctx
+    ngx.req.read_body()
+
+    local options = {
+      always_forward_body = true,
+      share_all_vars      = true,
+      method              = HTTP_METHODS[ngx.req.get_method()],
+      ctx                 = ctx,
+    }
+
+    local res = ngx.location.capture("/kong_buffered_http", options)
+    if res.truncated then
+      ngx.status = 502
+      return kong_error_handlers(ngx)
+    end
+
+
+    -- from response.exit
+    local core = kong.ctx.core
+    core.buffered_status = res.status
+    core.buffered_headers = res.header
+    core.buffered_body = res.body
+
+
+    -- from header_filter (s/HEADER_FILTER/BUFFERED_RESPONSE/)
+
+    if not ctx.KONG_PROCESSING_START then
+      ctx.KONG_PROCESSING_START = ngx.req.start_time() * 1000
+    end
+
+    if not ctx.workspace then
+      ctx.workspace = kong.default_workspace
+    end
+
+    if not ctx.KONG_BUFFERED_RESPONSE_START then
```

Comment:
as the phase is just `response` I propose (perhaps later on we can enable this without buffering, e.g. make a patch to nginx and make it real or something):

```suggestion
    if not ctx.KONG_RESPONSE_START then
```